rbi/core/kernel.rbi

```diff
@@ -1913,10 +1913,9 @@ module Kernel
   # #<S name="dave", state="TX">
   # ```
   sig do
-    params(
-        arg0: BasicObject,
-    )
-    .returns(NilClass)
+    type_parameters(:T)
+      .params(arg0: T.type_parameter(:T))
+      .returns(T.type_parameter(:T))
```

Comment:
There is a TODO up on line 1889 relevant to this

iirc the last time we touched this, we left it like this because it was tricky to try to write a signature that accounted for the fact that when given more than one argument, `p` returns an `Array`.